activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -107,6 +120,12 @@ def disconnect!
 
       private
 
+        def connected?
+          @connection && @connection.socket.present?
```

Comment:
This is sketchy, but we'll be able to clean that up as soon as https://github.com/brianmario/mysql2/pull/796 get merged.